packages/ui/package.json

```diff
@@ -66,7 +65,8 @@
     "type-fest": "^2.3.4"
   },
   "peerDependencies": {
-    "aws-amplify": "3.x.x || 4.x.x"
+    "aws-amplify": "3.x.x || 4.x.x",
+    "xstate": "^4.33.6"
```

Comment:
`xstate` is moved from a core dependency to a peer dependency of `@aws-amplify/ui`. This ensures the package will used the instance of `xstate` installed with the framework library of choice.